app/test/test-support/src/main/java/io/syndesis/test/container/s2i/S2iProjectBuilder.java

```diff
@@ -48,7 +50,21 @@ public Project build(IntegrationSource source) {
         SyndesisS2iAssemblyContainer syndesisS2iAssemblyContainer = new SyndesisS2iAssemblyContainer(integrationName, projectDir, imageTag);
         syndesisS2iAssemblyContainer.start();
 
-        Path fatJar = projectDir.resolve("target").resolve("project-0.1-SNAPSHOT.jar");
+        // The S2I assembly container result need to be copied to the local host
+        // to be used by S2I integration containers
+        Path target = projectDir.resolve("target");
+        Path fatJar = target.resolve("project-0.1-SNAPSHOT.jar");
+        String s2iAssemblyContainerSourcePath = "/tmp/src/target/project-0.1-SNAPSHOT.jar";
+        String localHostTargetDestination = fatJar.toAbsolutePath().toString();
+        try {
+            Files.createDirectories(target);
+            syndesisS2iAssemblyContainer.copyFileFromContainer(s2iAssemblyContainerSourcePath, localHostTargetDestination);
+        } catch (IOException e) {
+            throw new CannotCopyFromAssemblyContainerException(
+                String.format("Impossible to copy assembly fat jar from container path %s to local host path %s.", s2iAssemblyContainerSourcePath, localHostTargetDestination),
```

Comment:
I would suggest to have the error message as part of `CannotCopyFromAssemblyContainerException` and use container path and host path as constructor arguments there. Otherwise I do not see the benefit of having a dedicated exception class for this very specific case.

Reply:
I'd use `java.io.UncheckedIOException`

Reply:
Yes, I do prefer wrapping the exception into a standard one instead, thanks @zregvart.